### PR TITLE
imporve: ignore down server in extract_servers_from_upstram

### PR DESCRIPTION
**Pull request by starsz**
关联 issue：https://github.com/upyun/lua-resty-checkups/issues/48

### CODE REVIEW

**Comment by huangnauh** [comment]
> 关联 issue：#48

给函数加个参数吧， 比如 skip_down 之类， 保持现有的逻辑不变

**Comment by starsz** [comment]
OK

File: lib/resty/checkups/base.lua [hunk 1]
```diff
@@ -294,6 +295,10 @@ function _M.extract_servers_from_upstream(skey, cls)
     end
 
     for _, srv in ipairs(srvs) do
+        if ups_skip_down and srv.down then
+            goto continue
```
Comment (tokers) [inline lib/resty/checkups/base.lua]: LuaJIT's `goto` primitive cannot across the bound of new variables.

Comment (starsz) [reply lib/resty/checkups/base.lua]: OK, I got it.

Comment (starsz) [reply lib/resty/checkups/base.lua]: Thanks for your comment. But I found that [goto-scope](http://lua.2524044.n2.nabble.com/Clarification-of-goto-scope-td7241359.html)

According to the content of the email, I think I can use it like this.

Comment (tokers) [reply lib/resty/checkups/base.lua]: Well done.

**Pull request merged**